File: backend/app/routers/plot.py
from fastapi import APIRouter, Form
from fastapi.responses import FileResponse
from pathlib import Path
from app.utils.plotter import plot_train_results, plot_test_results

router = APIRouter()

@router.post("/plot/train")
async def plot_train(folder_path: str = Form(...)):
    folder = Path(folder_path)
    if not folder.exists():
        return {"error": "Folder not found"}
    try:
        plot_file = plot_train_results(folder)
        return FileResponse(plot_file, media_type="image/png")
    except Exception as e:
        return {"error": str(e)}

@router.post("/plot/test")
async def plot_test(folder_path: str = Form(...)):
    folder = Path(folder_path)
    if not folder.exists():
        return {"error": "Folder not found"}
    try:
        plot_file = plot_test_results(folder)
        return FileResponse(plot_file, media_type="image/png")
    except Exception as e:
        return {"error": str(e)}

File: frontend/src/api/plot.js
export async function plotTrain(folderPath) {
    const formData = new FormData();
    formData.append('folder_path', folderPath);
  
    const response = await fetch('/plot/train', {
      method: 'POST',
      body: formData,
    });
  
    if (response.ok) {
      return URL.createObjectURL(await response.blob());
    } else {
      console.error('Error plotting train results');
      return null;
    }
  }
  
  export async function plotTest(folderPath) {
    const formData = new FormData();
    formData.append('folder_path', folderPath);
  
    const response = await fetch('/plot/test', {
      method: 'POST',
      body: formData,
    });
  
    if (response.ok) {
      return URL.createObjectURL(await response.blob());
    } else {
      console.error('Error plotting test results');
      return null;
    }
  }
  